src/__experimental__/components/textarea/textarea.component.js

```diff
@@ -0,0 +1,172 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import I18n from 'i18n-js';
+import { validProps } from '../../../utils/ether';
+import { InputPresentation } from '../input';
+import FormField from '../form-field';
+import CharacterCount from './character-count';
+import TextareaInput from './textarea-input.component';
+
+const i18nNumberOpts = { precision: 0 };
+
+class Textarea extends React.Component {
+  static propTypes = {
+    /** Character limit of the textarea */
+    characterLimit: PropTypes.string,
+    /** Type of the icon that will be rendered next to the input */
+    children: PropTypes.node,
+    /** The visible width of the text control, in average character widths */
+    cols: PropTypes.number,
+    /** Adds disabled property */
+    disabled: PropTypes.bool,
+    /** Stop the user typing over the characterLimit */
+    enforceCharacterLimit: PropTypes.bool,
+    /** Allows the Textareas Height to change based on user input */
+    expandable: PropTypes.bool,
+    label: PropTypes.string,
+    onChange: PropTypes.func,
+    placeholder: PropTypes.string,
+    /** Adds readOnly property */
+    readOnly: PropTypes.bool,
+    /** The number of visible text lines for the control */
+    rows: PropTypes.number,
+    /** One of type of size to apply to the textarea */
+    size: PropTypes.oneOf(['small', 'medium', 'large']),
+    /** The value of the Textarea */
+    value: PropTypes.string,
+    /** Whether to display the character count message in red */
+    warnOverLimit: PropTypes.bool
+  }
+
+  static defaultProps = {
+    disabled: false,
+    expandable: false,
+    enforceCharacterLimit: true,
+    readOnly: false,
+    warnOverLimit: false
+  }
+
+  // Minimum height of the textarea
+  minHeight = 0;
+
+  /**
+   * A lifecycle method that is called after initial render.
+   * Allows access to refs and DOM to set expandable variables
+   */
+  componentDidMount() {
+    if (this.props.expandable) {
+      window.addEventListener('resize', this.expandTextarea);
+      // Set the min height to the initially rendered height.
+      // Without minHeight expandable textareas will only have
+      // one line when no content is present.
+      this.minHeight = this._input.clientHeight;
+
+      this.expandTextarea();
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.props.expandable) {
+      window.removeEventListener('resize', this.expandTextarea);
+    }
+  }
+
+  componentDidUpdate() {
+    const { expandable } = this.props;
+
+    if (expandable) {
+      this.expandTextarea();
+    }
```

Comment:
maybe `expandable && this.expandTextarea()` ?

Reply:
Could you tell me any reason why it will be better? I think it's more readable.

Reply:
I think one line is more readable ;)